datastream-server-restli/src/test/java/com/linkedin/datastream/server/TestCoordinator.java

```diff
@@ -230,6 +254,32 @@ public void initializeDatastream(Datastream stream, List<Datastream> allDatastre
     coordinator.stop();
   }
 
+  @Test
+  public void testConnectorGracefullyShutdown() throws Exception {
+    String testCluster = "testConnectorGracefullyShutdown";
+    String testConnectorType = "testConnectorType";
+    AtomicBoolean eventProcessed = new AtomicBoolean(false);
```

Comment:
I think using a `CountDownLatch` may be more appropriate for this purpose; we do not really need the semantics of an atomic variable here.